graphene/src/layers/style/mod.rs

```diff
@@ -26,27 +30,108 @@ impl Default for ViewMode {
 	}
 }
 
+/// A gradient fill.
+///
+/// Contains the start and end points, along with the colours at varying points along the length.
 #[repr(C)]
-#[derive(Debug, Clone, Copy, PartialEq, Default, Serialize, Deserialize)]
-pub struct Fill {
-	color: Color,
+#[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
+pub struct Gradient {
+	pub start: DVec2,
+	pub end: DVec2,
+	pub transform: DAffine2,
+	pub positions: Vec<(f64, Color)>,
+	uuid: u64,
+}
+impl Gradient {
+	/// Constructs a new gradient with the colours at 0 and 1 specified.
+	pub fn new(start: DVec2, start_colour: Color, end: DVec2, end_colour: Color, transform: DAffine2, uuid: u64) -> Self {
+		Gradient {
+			start,
+			end,
+			positions: vec![(0., start_colour), (1., end_colour)],
```

Comment:
Would you mind doing a global find and replace for `colour` ->` color`? I again apologize that it's necessary for the purpose of consistency even though your spelling is entirely correct and I'm basically asking you to change it to the wrong spelling 😝. I'm also happy to make the change if you prefer.